Typescript cannot find type declaration file when using import

When using typescript's node16 module and 'module' entry points typescript cannot find iter-tools's type definitions. 

### Setup
package.json
```
{
  "dependencies": {
    "@types/node": "^18.11.18",
    "iter-tools": "^7.5.0",
    "typescript": "^4.9.4"
  }
}
```
tsconfig.json
```
{
  "compilerOptions": {
    "strict": true,
    "target": "ES2022",
    "module": "node16",
    "moduleResolution": "node16",
  }
}
```
index.cts
```
import {asyncMap} from 'iter-tools'
```
index.mts
```
import {asyncMap} from 'iter-tools'
```

```
$ npx tsc --noEmit
index.mts:1:24 - error TS7016: Could not find a declaration file for module 'iter-tools'. 'test/node_modules/iter-tools/index.mjs' implicitly has an 'any' type.
  If the 'iter-tools' package actually exposes this module, try adding a new declaration (.d.ts) file containing `declare module 'iter-tools';`

1 import {asyncMap} from 'iter-tools'
```

--- a/package.es.json
+++ b/package.es.json
@@ -5,11 +5,13 @@
   "exports": {
     ".": {
       "import": "./index.mjs",
-      "require": "./index.js"
+      "require": "./index.js",
+      "types": "./index.d.ts"
     },
     "./methods/*": {
       "import": "./methods/*.mjs",
-      "require": "./methods/*.js"
+      "require": "./methods/*.js",
+      "types": "./methods/*.d.ts"
     },
     "./__methods/*": {
       "require": "./__methods/*.js"

--- a/package.es5.json
+++ b/package.es5.json
@@ -6,11 +6,13 @@
   "exports": {
     ".": {
       "import": "./index.mjs",
-      "require": "./index.js"
+      "require": "./index.js",
+      "types": "./index.d.ts"
     },
     "./methods/*": {
       "import": "./methods/*.mjs",
-      "require": "./methods/*.js"
+      "require": "./methods/*.js",
+      "types": "./methods/*.d.ts"
     },
     "./__methods/*": {
       "require": "./__methods/*.js"

--- a/package.json
+++ b/package.json
@@ -6,10 +6,12 @@
   "main": "src/index.js",
   "exports": {
     ".": {
-      "import": "./src/index.js"
+      "import": "./src/index.js",
+      "types": "./src/index.d.ts"
     },
     "./methods/*": {
-      "import": "./src/methods/*.js"
+      "import": "./src/methods/*.js",
+      "types": "./src/methods/*.d.ts"
     },
     "./__methods/*": {
       "import": "./src/__methods/*.js"

